Save Video Node "An iterator output was not assigned correctly"

**Information:**

- Chainner version: [Alpha v0.23.2]
- OS: [Windows 11 64 bit]

**Description**
As of [Alpha v0.23.2] an error is output when attempting to use a "save video" module via any input node method. Logs below show a "load video" -> "save video" error message.

**Logs**
An error occurred in a Save Video node:

An iterator output was not assigned correctly

Input values:
• Image Sequence: *unknown*
• Directory: *unknown*
• Video Name: *unknown*
• Video Format: *unknown*
• Encoder: *unknown*
• Video Preset: *unknown*
• Quality (CRF): *unknown*
• Additional parameters: *unknown*
• Additional parameters: *unknown*
• FPS: *unknown*
• Audio Stream: *unknown*
• Audio: *unknown*

Stack TraceTraceback (most recent call last):
  File "C:\Users\user\AppData\Local\chaiNNer\app-0.23.2\resources\src\process.py", line 415, in process
    return await self.__process(node)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\chaiNNer\app-0.23.2\resources\src\process.py", line 563, in __process
    inputs = await self.__gather_inputs(node)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\chaiNNer\app-0.23.2\resources\src\process.py", line 519, in __gather_inputs
    inputs.append(await self.__resolve_node_input(node_input))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\chaiNNer\app-0.23.2\resources\src\process.py", line 481, in __resolve_node_input
    return await self.__get_node_output(node_input.id, node_input.index)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\chaiNNer\app-0.23.2\resources\src\process.py", line 471, in __get_node_output
    assert value is not None, "An iterator output was not assigned correctly"
           ^^^^^^^^^^^^^^^^^
AssertionError: An iterator output was not assigned correctly

--- a/backend/src/process.py
+++ b/backend/src/process.py
@@ -505,18 +505,20 @@ class Executor:
         assigned_inputs = self.inputs.get(node.id)
         assert len(assigned_inputs) == len(node.data.inputs)
 
+        async def get_input_value(input_index: int, node_input: Input):
+            if input_index in ignore:
+                return None
+
+            if input_index in lazy:
+                return Lazy.from_coroutine(
+                    self.__resolve_node_input(assigned_inputs[input_index]), self.loop
+                )
+
+            return await self.__resolve_node_input(node_input)
+
         inputs = []
         for input_index, node_input in enumerate(assigned_inputs):
-            if input_index in ignore:
-                inputs.append(None)
-            if input_index in lazy:
-                inputs.append(
-                    Lazy.from_coroutine(
-                        self.__resolve_node_input(node_input), self.loop
-                    )
-                )
-            else:
-                inputs.append(await self.__resolve_node_input(node_input))
+            inputs.append(await get_input_value(input_index, node_input))
 
         return inputs
 

